iterator_test.go

```diff
@@ -2722,6 +2724,182 @@ func BenchmarkSeekPrefixTombstones(b *testing.B) {
 	}
 }
 
+func waitForCompactions(d *DB) {
+	d.mu.Lock()
+	// NB: Wait for table stats because some compaction types rely
+	// on table stats to be collected.
+	d.waitTableStats()
+	for d.mu.compact.compactingCount > 0 {
+		d.mu.compact.cond.Wait()
+		d.waitTableStats()
+	}
+	d.mu.Unlock()
+}
+
+// BenchmarkPointDeletedSwath benchmarks iterator operations on large-ish
+// (hundreds of MBs) databases containing broad swaths of keys removed by point
+// tombstones.
+func BenchmarkPointDeletedSwath(b *testing.B) {
+	const maxKeyLen = 5
+	ks := testkeys.Alpha(maxKeyLen)
+
+	opts := func() *Options {
+		return (&Options{
+			DisableWAL:         true,
```

Comment:
We shouldn't disable the WAL if the test is frequently closing/opening new DBs while ensuring saved state stays on the same underlying FS. We're already doing NoSync everywhere, so let's at least keep the WAL.